diff --git a/drivers/clk/renesas/r9a07g044-cpg.c b/drivers/clk/renesas/r9a07g044-cpg.c
index c21efb04634e..589b76be1078 100644
--- a/drivers/clk/renesas/r9a07g044-cpg.c
+++ b/drivers/clk/renesas/r9a07g044-cpg.c
@@ -129,7 +129,7 @@ static const char * const sel_gpu2[] = { ".pll6", ".pll3_div2_2" };
 static const u32 mtable_sdhi[] = { 1, 2, 3 };
 
 static const struct {
-	struct cpg_core_clk common[53];
+	struct cpg_core_clk common[54];
 #ifdef CONFIG_CLK_R9A07G054
 	struct cpg_core_clk drp[1];
 #endif
@@ -178,6 +178,7 @@ static const struct {
 
 		/* Core output clk */
 		DEF_DIV("I", R9A07G044_CLK_I, CLK_PLL1, DIVPL1A, dtable_1_8),
+		DEF_DIV("I2", R9A07G044_CLK_I2, CLK_PLL3_DIV2_4, DIVPL3CLK200FIX, dtable_1_32),
 		DEF_DIV("P0", R9A07G044_CLK_P0, CLK_PLL2_DIV2_8, DIVPL2A, dtable_1_32),
 		DEF_FIXED("P0_DIV2", R9A07G044_CLK_P0_DIV2, R9A07G044_CLK_P0, 1, 2),
 		DEF_FIXED("TSU", R9A07G044_CLK_TSU, CLK_PLL2_DIV2_10, 1, 1),
