diff --git a/drivers/clk/renesas/r9a07g043-cpg.c b/drivers/clk/renesas/r9a07g043-cpg.c
index 0777f3475b2b..362b7085191b 100644
--- a/drivers/clk/renesas/r9a07g043-cpg.c
+++ b/drivers/clk/renesas/r9a07g043-cpg.c
@@ -237,6 +237,10 @@ static struct rzg2l_mod_clk r9a07g043_mod_clks[] = {
 				0x56C, 0, 0),
 	DEF_MOD("lcdc_clkd",	R9A07G043_LCDC_CLK_D, R9A07G043_CLK_M3,
 				0x56C, 1, 0),
+	DEF_MOD("cm33_clkin",   R9A07G043_CM33_CLKIN, R9A07G043_CLK_I2,
+				0x504, 0, 0),
+	DEF_MOD("cm33_tsclk",   R9A07G043_CM33_TSCLK, R9A07G043_OSCCLK,
+				0x504, 1, 0),
 	DEF_MOD("ssi0_pclk",	R9A07G043_SSI0_PCLK2, R9A07G043_CLK_P0,
 				0x570, 0, MSTOP(MCPU1_MSTOP, BIT(10))),
 	DEF_MOD("ssi0_sfr",	R9A07G043_SSI0_PCLK_SFR, R9A07G043_CLK_P0,
@@ -366,6 +370,9 @@ static struct rzg2l_reset r9a07g043_resets[] = {
 	DEF_RST(R9A07G043_ADC_PRESETN, 0x8a8, 0),
 	DEF_RST(R9A07G043_ADC_ADRST_N, 0x8a8, 1),
 	DEF_RST(R9A07G043_TSU_PRESETN, 0x8ac, 0),
+	DEF_RST(R9A07G043_CM33_NPORESET, 0x804, 0),
+	DEF_RST(R9A07G043_CM33_NSYSRESET, 0x804, 1),
+	DEF_RST(R9A07G043_CM33_MISCRESETN, 0x804, 2),
 };
 
 static const unsigned int r9a07g043_crit_mod_clks[] __initconst = {
@@ -389,11 +396,11 @@ const struct rzg2l_cpg_info r9a07g043_cpg_info = {
 	/* Module Clocks */
 	.mod_clks = r9a07g043_mod_clks,
 	.num_mod_clks = ARRAY_SIZE(r9a07g043_mod_clks),
-	.num_hw_mod_clks = R9A07G043_TSU_PCLK + 1,
+	.num_hw_mod_clks = R9A07G043_CM33_TSCLK + 1,
 
 	/* Resets */
 	.resets = r9a07g043_resets,
-	.num_resets = R9A07G043_TSU_PRESETN + 1, /* Last reset ID + 1 */
+	.num_resets = R9A07G043_CM33_MISCRESETN + 1, /* Last reset ID + 1 */
 
 	.has_clk_mon_regs = true,
 };
diff --git a/include/dt-bindings/clock/r9a07g043-cpg.h b/include/dt-bindings/clock/r9a07g043-cpg.h
index 27e232733096..fc8d024e729b 100644
--- a/include/dt-bindings/clock/r9a07g043-cpg.h
+++ b/include/dt-bindings/clock/r9a07g043-cpg.h
@@ -108,6 +108,8 @@
 #define R9A07G043_ADC_ADCLK		76
 #define R9A07G043_ADC_PCLK		77
 #define R9A07G043_TSU_PCLK		78
+#define R9A07G043_CM33_CLKIN            79
+#define R9A07G043_CM33_TSCLK            80
 
 /* R9A07G043 Resets */
 #define R9A07G043_CA55_RST_1_0		0	/* RZ/G2UL Only */
@@ -180,5 +182,8 @@
 #define R9A07G043_ADC_PRESETN		67
 #define R9A07G043_ADC_ADRST_N		68
 #define R9A07G043_TSU_PRESETN		69
+#define R9A07G043_CM33_NPORESET         70
+#define R9A07G043_CM33_NSYSRESET        71
+#define R9A07G043_CM33_MISCRESETN       72
 
 #endif /* __DT_BINDINGS_CLOCK_R9A07G043_CPG_H__ */
